### ISSUE TITLE
Add way to disable/configure progress indicator

### ISSUE THREAD

**Issue by marcushellberg** [vaadin/hilla#2405]
### Describe your motivation

I'm already displaying progress on a component level for long-running endpoint calls. The app UX is deteriorated by the app-level progress indicator flashing during these operations. 

### Describe the solution you'd like

I would like a way to configure or disable the progress indicator when I'm already handling progress indication myself.

### Describe alternatives you've considered

Hiding it with CSS

### Additional context

As always, @TatuLund has a workaround https://vaadin.com/forum/t/how-do-i-disable-the-hilla-loading-indicator/166223/3 :)

**Comment by Legioth** [comment]
I would spontaneously think of two different approaches for this feature: based on the called method or based on the invocation.

If it's based on the method, then the indicator would never be shown for a specific method and you would thus have to expose multiple methods if you sometimes want to show the indicator and sometimes not.
```java
@BrowserCallable
public class HelloService {
  @BackgroundTask
  public String sayHelloWithoutProgressIndicator(String name) {
    return "Hello " + name
  }
  
  public String sayHelloWithProgressInidicator(String name) {
    return "Hello " + name    
  }
}
```

If it's based on the invocation, then we could overload the optional `EndpointRequestInit` parameter that is already there for the abort signal. This does instead add some boilerplate to every invocation (unless you curry it).
```typescript
const greeting = await HelloService.sayHello(name, {showProgress: false});
```

Any insights on which way would be preferable?

**Comment by marcushellberg** [comment]
Regardless of the annotation on the Java method, I would like to have the option to disable it on a per-call basis (the latter example). It's also more explicit to make you consider that you're actually indicating progress in some other way, whereas it would be easy to end up in a situation where one developer adds the backend annotation and another uses it in the frontend without noticing that and the UX suffers.

**Comment by Legioth** [comment]
Now when looking at this again, I'm preferring the `EndpointRequestInit` approach since that leaves the code change on the "UI side" of the implementation which is also where the alternative progress indicator is implemented.

If we assume that most endpoints methods are only called from one place in the code, then the amount of code is also comparable. Furthermore, adding the parameter to an existing type is slightly better for discoverability since you already have a natural entry point for spontaneously encountering the parameter whereas an annotation is something you just have to know about.

**Comment by marcushellberg** [comment]
Agreed. I think that giving control to the person/place calling the endpoint will give most flexibility for the implementation.

**Comment by platosha** [comment]
Let us not introduce Java annotations yet.

**Comment by cromoteca** [comment]
Workaround for now: use `Flux` and `Mono`.
